ambuild2/frontend/base_generator.py

```diff
@@ -0,0 +1,48 @@
+# vim: set ts=8 sts=4 sw=4 tw=99 et:
+#
+# This file is part of AMBuild.
+#
+# AMBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# AMBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with AMBuild. If not, see <http://www.gnu.org/licenses/>.
+
+class BaseGenerator(object):
+    def __init__(self, cm):
+        self.cm = cm
+
+    @property
+    def backend(self):
+        raise Exception('Must be implemented!')
```

Comment:
If we were using pylint for linting, you'll be chewed out for using ```Exception```
```suggestion
        raise NotImplementedError('Must be implemented!')
```
Infact, we can just turn this into an instance of ```ABCMeta``` no?
Though to do that, we require the "six" module.